internal/plugin/plugin.go

```diff
@@ -93,12 +95,39 @@ func NewPluginClient(connector string, version string, logger hclog.Logger) (Plu
 		return nil, fmt.Errorf("the plugin (%s) doesn't correctly implement the necessary interfaces", connector)
 	}
 
+	currentVersion := version.CurrentCliVersionInformation{}
+	if _, err = isPluginCompatibleWithCli(&currentVersion, is, logger); err != nil {
+		logger.Error("Plugin not compatible: %s", err.Error())
+		return nil, err
+	}
+
 	info := is.PluginInfo()
 	logger.Debug("Using plugin: " + info.String())
 
 	return pci, nil
 }
 
+func isPluginCompatibleWithCli(currentCliInformation version.CliVersionInformation, pluginInfo plugin2.Info, logger hclog.Logger) (bool, error) {
+	if currentCliInformation.GetCliVersion().Equal(version.DevVersion) {
+		logger.Warn("Skip plugin compatibility check for dev version")
+		return true, nil
+	}
+
+	//Check if current CLI version is compatible with CLI version used at plugin compile time (current version > plugin compiled version)
```

Comment:
Comment here isn't accurate. 
The CLI with which the plugin was compiled should be between the pluginCliMinimalVersion and current version.